powerdnsadmin/routes/index.py

```diff
@@ -1056,8 +1056,12 @@ def create_group_to_account_mapping():
 
 
 def handle_account(account_name, account_description=""):
+    if Setting().get('account_name_extra_chars'):
+        char_list = "abcdefghijklmnopqrstuvwxyz0123456789_-."
+    else:
+        char_list = "abcdefghijklmnopqrstuvwxyz0123456789"
     clean_name = ''.join(c for c in account_name.lower()
-                         if c in "abcdefghijklmnopqrstuvwxyz0123456789")
+                         if c in char_list)
     if len(clean_name) > Account.name.type.length:
```

Comment:
> in oidc group import, the check is done the same way in the handle_account function (updated). Tn hough it logs an error "Name truncated" it's still created

This actually checks against max length of the string column in the DB model, so it doesn't have much to do with this PR, but it nevertheless should be moved into `models.account.Account` on this occasion as well.

Reply:
Hi,

I don't agree with your comment, as `clean_name` is the formatted name from the character_list, and is used later to check/create the account.

Look [here](https://github.com/PowerDNS-Admin/PowerDNS-Admin/blob/23dcef764c37f17eb8ba3e21b4d924832ffa4efc/powerdnsadmin/routes/index.py#L1068-L1074) for the snippet. This is why I updated the code here as well. This would change the creation because the search would return no match with the new extended name, leading in a new account creation/association, but without updating the zones ownership.

Anyway, I'll use the model method instead.

Reply:
Sorry if I was unclear, I was only referring to [these](https://github.com/PowerDNS-Admin/PowerDNS-Admin/blob/23dcef764c37f17eb8ba3e21b4d924832ffa4efc/powerdnsadmin/routes/index.py#L1065-L1067) three lines, which lend themselves to being moved into the `Account` class as well on this occasion. Emitting an error log line if the account name (sanitized or not) will be truncated before written into the DB because it doesn't fit into max column character length shouldn't in itself have much to do with this PR.

Reply:
After looking at the code, I have more questions. There is something I don't get. I moved the sanitize function in the Account model, with the length check as requested, which just logs an error. Running Account.query.filter(account_name == something_over_40_chars) like it is done should return no value, triggering a creation that would fail because the name would be duplicate ? Or the filter is intelligent enough to truncate the requested name to match the field length ?

I could also see that in several places, when the function checks if the Account exists, it's done with the provided name. Either a call to  Account.query.filter sanitizes the name, which I expect it does not, and it's fine, or it does not and the check is always negative, but the account creation should then fail.

You can answer that I can run test to see how it behaves, but another pov is welcome.

The underlying question is: should we truncate the name in the sanitize function either, and filter by the sanitized name instead ?

Reply:
The test showed what I expected: The check don't match but the creation fails with a duplicate. 

I've updated the code to force the character limit in the sanitize function and the checks to compare the sanitized version, in both API, OIDC and Azure auto-provisioning.

Let me know if it suits your expectations and if the syntax style is good to you.

Reply:
Concerning LDAP, I'm not sure if we must enforce the sanitized name of the account existence check.